src/app.ts

```diff
@@ -40,3 +45,76 @@ export default async function newApp(): Promise<Express> {
   }
   return app;
 }
+
+function registerRoutes(app: Express): void {
+  app.get(
+    "/public-administrations",
+    [
+      query("search")
+        .not()
+        .isEmpty()
+        .withMessage("a value is required")
+        .isLength({ min: 3 })
+        .withMessage("value must have at least 3 characters")
+    ],
+    async (req: Request, res: Response) => {
+      const validationErrors = validationResult(req);
+      if (!validationErrors.isEmpty()) {
+        return res.status(400).json(validationErrors.array());
+      }
+      const searchString = req.query.search;
+      const requestUrl =
+        "https://elasticsearch.developers.italia.it/indicepa/_search";
+      const requestBody = {
+        query: {
+          bool: {
+            should: [
+              {
+                nested: {
+                  path: "office",
+                  query: {
+                    multi_match: {
+                      fields: ["office.code", "office.description"],
+                      operator: "and",
+                      query: searchString
+                    }
+                  }
+                }
+              },
+              {
+                multi_match: {
+                  fields: ["ipa", "description"],
+                  operator: "and",
+                  query: searchString
+                }
+              }
+            ]
+          }
+        }
+      };
+      try {
+        const searchResponse = await axios.post<IIpaSearchResponseBody>(
```

Comment:
you should use the [elasticsearch client](https://www.elastic.co/blog/new-elasticsearch-javascript-client-released) instead of doing raw HTTP request - less code + you get typescript support